e2e/registered-resources.bats

```diff
@@ -0,0 +1,388 @@
+
+  # renamed
+  run_otdfctl_reg_res update --id "$created_id" --name test_renamed_rr
+    assert_success
+    assert_line --regexp "Id.*$created_id"
+    assert_line --regexp "Name.*test_renamed_rr"
+    refute_output --regexp "Name.*test_update_rr"
+
+  # cleanup
+  run_otdfctl_reg_res delete --id $created_id --force
+}
+
+@test "Delete registered resource - Good" {
+  # setup a resource to delete
+  run_otdfctl_reg_res create --name test_delete_rr
+  created_id=$(echo "$output" | grep Id | awk -F'│' '{print $3}' | xargs)
+
+  run_otdfctl_reg_res delete --id "$created_id" --force
+    assert_success
+}
+
+@test "Delete registered resource - Bad" {
+  # no id
+  run_otdfctl_reg_res delete
+    assert_failure
+    assert_output --partial "Flag '--id' is required"
+
+  # invalid id
+  run_otdfctl_reg_res delete --id 'not_a_uuid'
+    assert_failure
+    assert_output --partial "must be a valid UUID"
+}
+
+# Tests for registered resource values
+
+@test "Create a registered resource value - Good" {
+  # simple
+  run_otdfctl_reg_res_values create --resource-id "$RR_ID" --value test_create_rr_val
+    assert_output --partial "SUCCESS"
+    assert_line --regexp "Value.*test_create_rr_val"
+    assert_output --partial "Id"
+    assert_output --partial "Created At"
+    assert_line --partial "Updated At"
+  created_id_simple=$(echo "$output" | grep Id | awk -F'│' '{print $3}' | xargs)
+
+  # with action attribute values
+  run_otdfctl_reg_res_values create --resource-id "$RR_ID" --value test_create_rr_val_with_action_attr_vals --action-attribute-value "\"$READ_ACTION_ID;$ATTR_VAL_1_FQN\"" --action-attribute-value "\"$CUSTOM_ACTION_NAME;$ATTR_VAL_2_ID\"" --json
+    assert_success
+    [ "$(echo "$output" | jq -r '.id')" != "" ]
+    [ "$(echo "$output" | jq -r '.value')" = "test_create_rr_val_with_action_attr_vals" ]
+    [ "$(echo "$output" | jq -r '.action_attribute_values[0].action.id')" = "$READ_ACTION_ID" ]
+    [ "$(echo "$output" | jq -r '.action_attribute_values[0].action.name')" = "$READ_ACTION_NAME" ]
+    [ "$(echo "$output" | jq -r '.action_attribute_values[0].attribute_value.id')" = "$ATTR_VAL_1_ID" ]
+    [ "$(echo "$output" | jq -r '.action_attribute_values[0].attribute_value.fqn')" = "$ATTR_VAL_1_FQN" ]
+    [ "$(echo "$output" | jq -r '.action_attribute_values[1].action.id')" = "$CUSTOM_ACTION_ID" ]
+    [ "$(echo "$output" | jq -r '.action_attribute_values[1].action.name')" = "$CUSTOM_ACTION_NAME" ]
+    [ "$(echo "$output" | jq -r '.action_attribute_values[1].attribute_value.id')" = "$ATTR_VAL_2_ID" ]
+    [ "$(echo "$output" | jq -r '.action_attribute_values[1].attribute_value.fqn')" = "$ATTR_VAL_2_FQN" ]
+  created_id_with_action_attr_vals=$(echo "$output" | jq -r '.id')
+
+  # cleanup
+  run_otdfctl_reg_res_values delete --id $created_id_simple --force
+  run_otdfctl_reg_res_values delete --id $created_id_with_action_attr_vals --force
+}
+
+@test "Create a registered resource value - Bad" {
```

Comment:
Can you please add test cases for your delimiter `;` that's good/bad so we can detect regressions if anything changes if anything around the delimiter changes going forward?